packages/databases-collections/src/components/create-namespace-modal.spec.tsx

```diff
@@ -0,0 +1,332 @@
+          fireEvent.click(submitButton);
+        });
+
+        it('trims the white spaces on submit', async () => {
+          await waitFor(() => {
+            expect(createNamespaceStub).to.have.been.calledOnceWithExactly(
+              'foo.baz',
+              {}
+            );
+          });
+        });
+      }
+    );
+  });
+
+  context(
+    'Create collection - the database has extra whitespaces',
+    function () {
+      beforeEach(async function () {
+        const Plugin = CreateNamespacePlugin.withMockServices(mockServices);
+
+        render(<Plugin> </Plugin>);
+        appRegistry.emit(
+          'open-create-collection',
+          {
+            database: '  foo',
+          },
+          { connectionId: '12345' }
+        );
+
+        await waitFor(() =>
+          screen.getByRole('heading', { name: 'Create Collection' })
+        );
+      });
+
+      afterEach(function () {
+        sandbox.resetHistory();
+        cleanup();
+      });
+
+      context(
+        'when the user submitted the collection creation form with extra whitespaces',
+        () => {
+          beforeEach(() => {
+            const submitButton = screen.getByRole('button', {
+              name: 'Create Collection',
+            });
+            const input = screen.getByRole('textbox', {
+              name: 'Collection Name',
+            });
+            fireEvent.change(input, { target: { value: '  baz  ' } });
+            fireEvent.click(submitButton);
+          });
+
+          it('trims the white spaces on submit - but only for the collection', async () => {
+            await waitFor(() => {
+              expect(createNamespaceStub).to.have.been.calledOnceWithExactly(
+                '  foo.baz',
+                {}
+              );
+            });
+          });
+        }
+      );
+    }
+  );
+
+  context('Create database + collection', function () {
+    beforeEach(async function () {
+      const Plugin = CreateNamespacePlugin.withMockServices(mockServices);
+
+      render(<Plugin> </Plugin>);
+      appRegistry.emit('open-create-database', { connectionId: '12345' });
+
+      await waitFor(() =>
+        screen.getByRole('heading', { name: 'Create Database' })
+      );
+    });
+
+    afterEach(function () {
+      sandbox.resetHistory();
+      cleanup();
+    });
+
+    it('renders the correct text on the submit button', () => {
+      const submitButton = screen.getByTestId('submit-button');
+      expect(submitButton.textContent).to.equal('Create Database');
+    });
+
+    it('button is disabled when the input is empty', () => {
+      const submitButton = screen.getByTestId('submit-button');
+      const dbInput = screen.getByTestId('database-name');
+      const collInput = screen.getByTestId('collection-name');
+      expect(submitButton.getAttribute('aria-disabled')).to.equal('true');
+
+      fireEvent.change(dbInput, { target: { value: 'db1' } });
```

Comment:
Please use userEvent instead if possible, we're following testing library recs and moving away from fireEvent

Reply:
Thanks!